File: declined-php/src/DeclinedException.php
<?php

declare(strict_types=1);

namespace Declined;

final class DeclinedException extends \RuntimeException
{
    public function __construct(
        public readonly int $status,
        string $message,
        public readonly ?string $code = null,
    ) {
        parent::__construct($message);
    }
}

File: declined-java/src/main/java/io/declined/DeclinedException.java
package io.declined;

public class DeclinedException extends RuntimeException {
  private final int status;
  private final String code;

  public DeclinedException(int status, String message, String code) {
    super(message);
    this.status = status;
    this.code = code;
  }

  public int getStatus() {
    return status;
  }

  public String getCode() {
    return code;
  }
}
